src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -4037,6 +4037,13 @@ public override SyntaxNode SwitchSection(IEnumerable<SyntaxNode> expressions, IE
             return SyntaxFactory.SwitchSection(AsSwitchLabels(expressions), AsStatementList(statements));
         }
 
+        internal override SyntaxNode PatternSwitchSection(IEnumerable<SyntaxNode> labels, IEnumerable<SyntaxNode> statements)
```

Comment:
SwitchSection gets a list of expressions and turn them to simple case clauses (constant pattern), PatternSwitchSection passes it as-is (notice we get "labels" and not "expressions").